### PR TITLE
SLCORE-646 Introduce explicit feature flag for DBD

### PR DESCRIPTION
**Pull request by jblievremont**

### CODE REVIEW

File: core/src/main/java/org/sonarsource/sonarlint/core/ConnectedSonarLintEngineImpl.java [hunk 1]
```diff
@@ -140,7 +141,8 @@ private PluginsLoadResult loadPlugins() {
     pluginsToLoadByKey.putAll(globalConfig.getEmbeddedPluginPathsByKey());
     Set<Path> plugins = new HashSet<>(pluginsToLoadByKey.values());
 
-    var config = new Configuration(plugins, globalConfig.getEnabledLanguages(), Optional.ofNullable(globalConfig.getNodeJsVersion()), globalConfig.getAllowedPluginIds());
+    var config = new Configuration(plugins, globalConfig.getEnabledLanguages(), Optional.ofNullable(globalConfig.getNodeJsVersion()),
+      DataflowBugDetection.getPluginAllowList(globalConfig.isDataflowBugDetectionEnabled()));
```
Comment (henryju) [inline core/src/main/java/org/sonarsource/sonarlint/core/ConnectedSonarLintEngineImpl.java:145]: I'm tempted to go one step further and pass the `isDataflowBugDetectionEnabled` boolean to the PluginLoader configuration (like enabledLanguages) and only internally decide which plugins to load.

Comment (sophio-japharidze-sonarsource) [inline core/src/main/java/org/sonarsource/sonarlint/core/ConnectedSonarLintEngineImpl.java:145]: Is this parameter still relevant with the change you made in PluginServiceImpl? Looks duplicated to me 🤔

Comment (jblievremont) [reply core/src/main/java/org/sonarsource/sonarlint/core/ConnectedSonarLintEngineImpl.java]: I tried to push the feature flag even further down; we still need it both on the engine side _and_ backend side AFAICT.

**Review by sophio-japharidze-sonarsource — APPROVED** [review]

File: commons/src/main/java/org/sonarsource/sonarlint/core/commons/DataflowBugDetection.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonarsource.sonarlint.core.commons;
+
+import java.util.Set;
+
+public class DataflowBugDetection {
```
Comment (henryju) [inline commons/src/main/java/org/sonarsource/sonarlint/core/commons/DataflowBugDetection.java:24]: Can't this file be moved to `plugin-commons` module?

File: its/tests/src/test/java/its/SonarQubeDeveloperEditionTests.java [hunk 3]
```diff
@@ -181,7 +181,7 @@ static void start() {
     backend = new SonarLintBackendImpl(newDummySonarLintClient());
     try {
       backend.initialize(
-        new InitializeParams(IT_CLIENT_INFO, new FeatureFlagsDto(false, true, false, false, false, true), sonarUserHome.resolve("storage"), sonarUserHome.resolve("workDir"),
+        new InitializeParams(IT_CLIENT_INFO, new FeatureFlagsDto(false, true, false, false, false, true, false), sonarUserHome.resolve("storage"), sonarUserHome.resolve("workDir"),
```
Comment (henryju) [inline its/tests/src/test/java/its/SonarQubeDeveloperEditionTests.java:184]: Do we have at least one IT to validate this feature? Or are we maybe waiting for changes on the DBD plugin side?

**Review by henryju — APPROVED** [review]
Looks very good to me. I just think we should put one class in a different module.

File: its/tests/src/test/java/its/SonarQubeDeveloperEditionTests.java (hunk 3 above)
Comment (jblievremont) [reply its/tests/src/test/java/its/SonarQubeDeveloperEditionTests.java:184]: Yup, writing an IT for this feature would be premature right now.

Comment (henryju) [reply its/tests/src/test/java/its/SonarQubeDeveloperEditionTests.java:184]: Maybe create a SLCORE ticket to not forget

Comment (jblievremont) [reply its/tests/src/test/java/its/SonarQubeDeveloperEditionTests.java:184]: Created [SLCORE-660](https://sonarsource.atlassian.net/browse/SLCORE-660), thanks!

**Pull request merged**